src/codec/builtin/FixSizedTypesCodec.ts

```diff
@@ -43,6 +73,29 @@ export class FixSizedTypesCodec {
         return new Long(low, high);
     }
 
+    static encodeNumberAsLong(buffer: Buffer, offset: number, value: any): void {
```

Comment:
As this method assumes non-negative numbers only, it should be renamed into smth like `encodeNonNegativeNumberAsLong` (or smth like that) to avoid confusion in future. It might be also a good idea to validate the input and throw if it's negative.

Alternatively, this method should be updated to support negative numbers.

Reply:
ok, I chose to use `encodePositiveNumber` to not have a double negative in the name : "non" "negative"
I chose to throw `Error` also , please let me know if there is a more suitable error

Reply:
Non-negative is a bit more precise, as the method works fine for `0`. As for the error class, it's fine to use `Error` here, as users should never see this exception.

Reply:
ok, good point